optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/OptaWebExceptionHandler.java

```diff
@@ -26,12 +26,8 @@
 @ApplicationScoped
 public class OptaWebExceptionHandler implements ExceptionMapper<Throwable> {
 
-    private ExceptionDataMapper exceptionDataMapper;
-
     @Inject
-    public OptaWebExceptionHandler(ExceptionDataMapper exceptionDataMapper) {
-        this.exceptionDataMapper = exceptionDataMapper;
-    }
+    ExceptionDataMapper exceptionDataMapper;
```

Comment:
I would prefer if the fields stayed `private final`. It conveys an important piece of information to the reader: the field is initialized when an instance is created and can never ever change. In my opinion this reduces cognitive complexity because I can immediately see it is a dependency the never changes during the instance life cycle.

Reply:
Can't since Quarkus need to create a proxy of this class as it is `@ApplicationScoped` (and Quarkus seems to ignore constructor `@Inject` points and end up attempting to create a default no-args constructor)

Reply:
I can't confirm this. Please revisit the constructor injection option. If it still doesn't work for you, please share the details.

Reply:
It works, but with a warning: `2021-02-22 12:36:19,581 WARN  [io.qua.res.com.dep.ResteasyCommonProcessor] (build-37) Classes annotated with @Provider should have a single, no-argument constructor, otherwise dependency injection won't work properly. Offending class is org.optaweb.employeerostering.OptaWebExceptionHandler`. ExceptionDataMapper isn't an interface and doesn't use CDI, so I can just create it in the constructor